sys/vm/vm_map.c

```diff
@@ -1576,12 +1608,122 @@ vm_map_entry_unlink(vm_map_t map, vm_map_entry_t entry,
 	if (root != NULL)
 		root->max_free = vm_size_max(max_free_left, max_free_right);
 	map->root = root;
+#ifdef CHERI_CAPREVOKE
+	if (entry->inheritance == VM_INHERIT_QUARANTINE) {
```

Comment:
Should we also make sure that `map->rev_entry` isn't left dangling? That is, assert that `map->rev_entry != entry`.